### PR TITLE
libmtp: 1.1.18 -> 1.1.19

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 144290` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnome.gnome-documents</li>
    <li>libsForQt512.elisa</li>
    <li>libsForQt512.kio-extras</li>
    <li>libsForQt512.phonon-backend-vlc</li>
    <li>libsForQt514.elisa</li>
    <li>libsForQt514.kio-extras</li>
    <li>libsForQt514.phonon-backend-vlc</li>
    <li>tribler</li>
    <li>xmonad_log_applet</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>almanah</li>
    <li>evolution</li>
    <li>evolution-ews</li>
    <li>evolutionWithPlugins</li>
    <li>gnome.gnome-control-center</li>
    <li>gnome.gnome-todo</li>
    <li>obs-studio-plugins.obs-ndi</li>
    <li>phosh</li>
  </ul>
</details>
<details>
  <summary>110 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>amarok</li>
    <li>arcan.all-wrapped</li>
    <li>arcan.arcan</li>
    <li>arcan.arcan-wrapped</li>
    <li>arcan.durden-wrapped</li>
    <li>arcan.pipeworld-wrapped</li>
    <li>arcan.prio-wrapped</li>
    <li>arcan.xarcan</li>
    <li>bubblemail</li>
    <li>calibre</li>
    <li>calls</li>
    <li>cantata</li>
    <li>chatty</li>
    <li>chrome-gnome-shell</li>
    <li>cinnamon.cinnamon-common</li>
    <li>cinnamon.cinnamon-control-center</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>clementine</li>
    <li>clementineUnfree</li>
    <li>dropbox-cli</li>
    <li>elementary-planner</li>
    <li>empathy</li>
    <li>evolution-data-server</li>
    <li>far2l</li>
    <li>feedreader</li>
    <li>folks</li>
    <li>gfbgraph</li>
    <li>gmtp</li>
    <li>gnome-online-accounts</li>
    <li>gnome-photos</li>
    <li>gnome-recipes</li>
    <li>gnome.cheese</li>
    <li>gnome.file-roller</li>
    <li>gnome.geary</li>
    <li>gnome.gnome-applets</li>
    <li>gnome.gnome-books</li>
    <li>gnome.gnome-boxes</li>
    <li>gnome.gnome-calendar</li>
    <li>gnome.gnome-contacts</li>
    <li>gnome.gnome-flashback</li>
    <li>gnome.gnome-initial-setup</li>
    <li>gnome.gnome-maps</li>
    <li>gnome.gnome-music</li>
    <li>gnome.gnome-notes</li>
    <li>gnome.gnome-online-miners</li>
    <li>gnome.gnome-panel</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-terminal</li>
    <li>gnome.gnome-tweaks</li>
    <li>gnome.gnome-user-share</li>
    <li>gnome.gvfs</li>
    <li>gnome.nautilus</li>
    <li>gnome.nautilus-python</li>
    <li>gnome.totem</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.night-theme-switcher</li>
    <li>grilo-plugins</li>
    <li>gvfs</li>
    <li>haruna</li>
    <li>jmtpfs</li>
    <li>kgx</li>
    <li>libgdata</li>
    <li>libmtp</li>
    <li>libsForQt5.elisa</li>
    <li>libsForQt5.kio-extras</li>
    <li>libsForQt5.phonon-backend-vlc</li>
    <li>libvlc</li>
    <li>libzapojit</li>
    <li>mailnagWithPlugins</li>
    <li>megaglest</li>
    <li>minitube</li>
    <li>mtpfs</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-nvfbc</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>openlpFull</li>
    <li>pantheon.elementary-calendar</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-mail</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-tasks</li>
    <li>pantheon.extra-elementary-contracts</li>
    <li>pantheon.file-roller</li>
    <li>pantheon.switchboard-plug-onlineaccounts</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>pympress</li>
    <li>python38Packages.python-vlc</li>
    <li>python39Packages.python-vlc</li>
    <li>qtscrobbler</li>
    <li>rabbitvcs</li>
    <li>reaper</li>
    <li>shotwell</li>
    <li>spaceFM</li>
    <li>strawberry</li>
    <li>tracker-miners</li>
    <li>vimix-gtk-themes</li>
    <li>vlc</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>

File: pkgs/development/libraries/libmtp/default.nix [hunk 1]
```diff
@@ -24,30 +31,26 @@ stdenv.mkDerivation rec {
...
-    libiconv
-  ];
+  buildInputs = [ libiconv ];
 
-  propagatedBuildInputs = [
-    libusb1
-  ];
+  propagatedBuildInputs = [ libusb1 ];
+
+  preConfigure = "./autogen.sh";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libmtp/default.nix:38]: This should be a multiline string.

File: pkgs/development/libraries/libmtp/default.nix [hunk 2]
```diff
@@ -24,30 +31,26 @@ stdenv.mkDerivation rec {
...
 
-  preConfigure = ''
-    ./autogen.sh
-  '';
+  configureFlags = [ "--with-udev=${placeholder "out"}/lib/udev" ];
```
Comment (sigprof) [inline pkgs/development/libraries/libmtp/default.nix:40]: Apparently this change moved the udev rules for libmtp from the `bin` to the `out` output, but `nixos/modules/services/desktops/gvfs.nix` still expects these rules to be in the `bin` output:
https://github.com/NixOS/nixpkgs/blob/fcb125441fdefeeea3e3f0a6dce725e7bce44fc7/nixos/modules/services/desktops/gvfs.nix#L57
Therefore the udev rules for libmtp do not get applied, which breaks access to MTP devices for users.

Should the udev rules be returned to the `bin` output, or should the `gvfs` module be adjusted to refer to `pkgs.libmtp.out` instead? (Note that plain `pkgs.libmtp` also refers to the `bin` output and therefore does not work in `services.udev.packages` at the moment.)

This problem also affects the NixOS 21.11 release.

Comment (lovesegfault) [reply pkgs/development/libraries/libmtp/default.nix:40]: Addressed in #152289, nice catch!

**Pull request merged**